internal/dag/builder.go

```diff
@@ -880,6 +881,26 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 	return routes
 }
 
+// determineSNI decides what the SNI should be on the request
```

Comment:
Some explanation of what happens where there should be no SNI on the request would be good.